src/main/java/org/ow2/proactive/catalog/report/HeadersBuilder.java

```diff
@@ -0,0 +1,128 @@
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import java.net.URL;
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.imageio.ImageIO;
+
+import org.apache.pdfbox.pdmodel.PDPage;
+import org.ow2.proactive.catalog.dto.CatalogObjectMetadata;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import be.quodlibet.boxable.BaseTable;
+import be.quodlibet.boxable.Cell;
+import be.quodlibet.boxable.HorizontalAlignment;
+import be.quodlibet.boxable.Row;
+import be.quodlibet.boxable.image.Image;
+
+
+@Component
+public class HeadersBuilder {
+
+    private static final String ACTIVEEON_BLUE = "#0E2C65";
+
+    private static final String ACTIVEEON_ORANGE = "#EE7939";
+
+    @Value("${pa.scheduler.url}")
+    private String schedulerUrl;
+
+    private static final String ACTIVEEON_LOGO = "/automation-dashboard/styles/patterns/AE-Logo.png";
+
+    private static final String MAIN_TITLE = "ProActive Catalog report";
+
+    public void createMainHeader(BaseTable table) throws IOException {
+        Row<PDPage> headerRow = table.createRow(15f);
+
+        URL url = new URL(schedulerUrl + ACTIVEEON_LOGO);
+        BufferedImage imageFile = ImageIO.read(url);
+        headerRow.createImageCell((100 / 8f), new Image(imageFile));
+
+        Cell<PDPage> cell = headerRow.createCell((100 / 8f) * 7, MAIN_TITLE);
+        cell.setFillColor(java.awt.Color.decode(ACTIVEEON_ORANGE));
+        cell.setTextColor(java.awt.Color.decode(ACTIVEEON_BLUE));
+        cell.setAlign(HorizontalAlignment.CENTER);
+
+        table.addHeaderRow(headerRow);
+    }
+
+    public void createInfoHeader(BaseTable table, Set<CatalogObjectMetadata> orderedObjectsPerBucket,
+            Optional<String> kind, Optional<String> contentType) {
+        Row<PDPage> infoHeaderRow = table.createRow(15f);
+
+        String infoHeaderMessage = "Objects Kind: " + kind.orElse("All") + ", Content Type: " +
+                                   contentType.orElse("All") + ", Bucket Number: " +
+                                   calculateTotalBuckets(orderedObjectsPerBucket) + ",  Object Number: " +
+                                   orderedObjectsPerBucket.size() + ", Generated: " + getNowDate();
+
+        infoHeaderRow.createCell(100, infoHeaderMessage);
+
+    }
+
+    private int calculateTotalBuckets(Set<CatalogObjectMetadata> orderedObjectsPerBucket) {
+        String currentBucketName = "";
+        int totalNumberOfBuckets = 0;
+
+        for (CatalogObjectMetadata catalogObject : orderedObjectsPerBucket) {
+
+            if (!currentBucketName.equals(catalogObject.getBucketName())) {
+                currentBucketName = catalogObject.getBucketName();
+                totalNumberOfBuckets++;
+            }
+
+        }
+
+        return totalNumberOfBuckets;
+    }
+
+    public void createEmptyTotalsHeader(Row<PDPage> totalsHeaderRow, int totalNumberOfBuckets) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)